### PR TITLE
Implementing the refactoring

### PR DESCRIPTION
**Pull request by miatrinity**

### CODE REVIEW

File: app/presenters/base_presenter.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class BasePresenter
+  def initialize(object, template)
+    @object = object
+    @template = template
+  end
+
+private
+
+  # presents :routine_flow
```
Comment (MiaSinek) [inline app/presenters/base_presenter.rb]: Remove comment

File: app/presenters/base_presenter.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  def self.presents(name)
+    define_method(name) do
+      @object
+    end
+  end
+
+  def h
+    @template
+  end
```
Comment (MiaSinek) [inline app/presenters/base_presenter.rb]: Is this needed at all? 🤔

File: app/presenters/routine_flow_presenter.rb [hunk 3]
```diff
@@ -1,14 +1,26 @@
...
-  include ViewCompatibilityPack
-
+class RoutineFlowPresenter < BasePresenter
+  presents :routine_flow
   delegate :routine, to: :routine_flow, prefix: false
 
-  def initialize(routine_flow)
-    @routine_flow = routine_flow
+  def title
+    routine_flow.title
```
Comment (MiaSinek) [inline app/presenters/routine_flow_presenter.rb]: Delegate to routine_flow

File: app/presenters/routine_flow_presenter.rb [hunk 4]
```diff
@@ -1,14 +1,26 @@
...
+  def reports_button
+    link_to 'Reports',
+            routine_routine_flow_reports_path(routine),
+            class: 'button is-large is-info is-fullwidth'
+  end
+
+  def routines_button
+    link_to 'Routines',
+            routines_path,
+            class: 'button is-large is-info is-fullwidth'
```
Comment (MiaSinek) [inline app/presenters/routine_flow_presenter.rb]: Are these presenting routine_flow? Where should we move so it semantically makes more sense? 🤔

**Review by MiaSinek — CHANGES_REQUESTED** [review]

File: app/presenters/base_presenter.rb (hunk 1 above)
Comment (miatrinity) [reply app/presenters/base_presenter.rb]: Okay!

File: app/presenters/base_presenter.rb (hunk 2 above)
Comment (MiaSinek) [reply app/presenters/base_presenter.rb:18]: Nope! 💪

Comment (miatrinity) [reply app/presenters/base_presenter.rb]: Nope! 💪

File: app/presenters/routine_flow_presenter.rb (hunk 3 above)
Comment (miatrinity) [reply app/presenters/routine_flow_presenter.rb]: Done

**Review by MiaSinek — APPROVED** [review]

**Pull request merged**